### ISSUE TITLE
Best practices for links between Gatsby static pages and client-only routes

### ISSUE THREAD

**Issue by stevenkkim** [gatsbyjs/gatsby#23385]
I've read the documentation Gatsby `<Link>`, client-only routes, and @reach/router.

I think I have a good understanding of how it works. However, I'm wondering whether there are recommendations for linking between static pages and client-only routes, in particular using Gatsby `<Link>` vs @reach/router `<Link>`

Let's say for example, that there are static pages at `/documentation/*`
And let's say there's an app with client-only routes at `/app/*`

There are 4 scenarios here:

1. static to static
(e.g. a link on `/documentation/feature-1` to `/documentation/feature-2`):
obviously should use Gatsby `<Link>` here.

2. static to client-only
(e.g. a link on `/documentation/feature-1` to `/app/feature-1`):
Should I use Gatsby `<Link>` or @reach/router `<Link>`? Does it matter? Since the destination is client-only, I'm guessing that pre-load for Gatsby `<Link>` won't work?

3. client-only to client-only
(e.g. a link on `/app/feature-1` to `/app/feature-2`):
I think I should use @reach/router `<Link>` for this. But could I use Gatsby `<Link>` here? Are there any benefits to doing so?

4. client-only to static
(e.g. a link on `/app/feature-1` to `/documentation/feature-1`):
In this case, I think should I use Gatsby `<Link>` since the destination is static and prefetching can happen.

Maybe the answer here is simple and I should just use Gatsby `<Link>` for everything, but I'm not sure if that's the right answer and what the implications are if any.

Also, it may be helpful to address this question in the Gatsby Link or client-only routes reference guides.

Thanks!

**Comment by herecydev** [comment]
With regards to 2, it will still prefetch the page that will handle the client-only routes (and therefore will appear faster). One advantage to using the gatsby `Link` is that it should remain a consistent API, in the future [I would expect](https://reacttraining.com/blog/reach-react-router-future/) a move to react-router. If you rely on @reach-router links then that'll have to be changed.

But really I wouldn't overworry (other than my 1st point), Gatsby `Link` is just a super-thin wrapper around @reach-router's `Link`

**Comment by stevenkkim** [comment]
Thanks for the advice. I'll go with Gatsby `Link` for everything for now.

**Comment by LekoArts** [comment]
Thank you for opening this!

Indeed, using Gatsby's Link for everything isn't harmful, for the 3rd case it would also make sense to only use your Routing choice (e.g. if you want to use react-router-dom).

We're marking this issue as answered and closing it for now but please feel free to comment here if you would like to continue this discussion. We also recommend heading over to our [communities](https://www.gatsbyjs.org/contributing/community/#where-to-get-support) if you have questions that are not bug reports or feature requests. We hope we managed to help and thank you for using Gatsby!

**Comment by stevenkkim** [comment]
@LekoArts thanks for pointing me in the direction of the communities/support page! I hadn't seen that before.

**Comment by MauriceOppenberger** [comment]
@LekoArts and @stevenkkim  thank you for this issue.  
I am experiencing an issue with 
>static to client-only
(e.g. a link on /documentation/feature-1 to /app/feature-1):
Should I use Gatsby <Link> or @reach/router <Link>? Does it matter? Since the destination is client-only, I'm guessing that pre-load for Gatsby <Link> won't work?

where the link to a client-route goes first to the 404 page, only after reloading the page it would work. 

thanks to this issue I have tried replacing my reach-router Link component with the gatsby Link component and the issue was resolved. 
I don't quite understand what the difference between the Link components is but apparently for case 2 the gatsby Link component works better

side note: the issue only happens in dev env
